### PR TITLE
[tests-only] Adjust pact interactions for sharingWithAttributesTest

### PR DESCRIPTION
**Pull request by saw-jan**
part of https://github.com/owncloud/owncloud-sdk/issues/709 and https://github.com/owncloud/owncloud-sdk/issues/644

adjustment for test suites:
- sharingWithAttributesTest

### CODE REVIEW

**Review by Talank — COMMENTED** [review]
LGTM

File: tests/pactHelper.js [hunk 1]
```diff
@@ -38,20 +38,19 @@ const ocsMeta = function (meta, status, statusCode, Message = null) {
...
+    .appendElement('uid_file_owner', '', MatchersV3.string(config.adminUsername))
+    .appendElement('displayname_file_owner', '', MatchersV3.string(config.adminUsername))
     .appendElement('path', '', fileTarget)
     .appendElement('file_target', '', fileTarget)
-    .appendElement('stime', '', Math.floor(Date.now() / 1000))
+    .appendElement('stime', '', MatchersV3.string(Math.floor(Date.now() / 1000)))
 
   if (shareType === 3) {
-    res.appendElement('url', '', config.owncloudURL + '/s/yrkoLeS33y1aTya')
+    res.appendElement('url', '', MatchersV3.string(config.owncloudURL + '/s/yrkoLeS33y1aTya'))
```
Comment (individual-it) [inline tests/pactHelper.js:53]: what about having a regular expression here? maybe we want to make sure it starts with `/s/`

File: tests/sharingWithAttributesTest.js [hunk 2]
```diff
@@ -4,40 +4,42 @@ describe('oc.shares', function () {
...
 
   const shareAttributesResponse = () => {
     const response = []
     for (let i = 0; i < shareAttributes.attributes.length; i++) {
-      response.push(`{
-      &quot;scope&quot;:&quot;${shareAttributes.attributes[i].scope}&quot;,
-      &quot;key&quot;:&quot;${shareAttributes.attributes[i].key}&quot;,
-      &quot;enabled&quot;:&quot;${shareAttributes.attributes[i].value}&quot;
-      }`)
+      response.push('{"scope":"' + shareAttributes.attributes[i].scope + '","key":"' + shareAttributes.attributes[i].key + '",' +
```
Comment (individual-it) [inline tests/sharingWithAttributesTest.js:32]: :+1: looks soo much nicer

File: tests/sharingWithAttributesTest.js [hunk 3]
```diff
@@ -46,10 +48,13 @@ describe('oc.shares', function () {
           '/ocs/v1.php/apps/files_sharing/api/v1/shares'
         ),
         headers: {
-          ...validAdminAuthHeaders,
+          authorization: getAuthHeaders(testUser, testUserPassword),
           ...applicationFormUrlEncoded
         },
-        body: `shareType=0&shareWith=${testUser}&path=%2F${testFile}&attributes%5B0%5D%5Bscope%5D=${shareAttributes.attributes[0].scope}&attributes%5B0%5D%5Bkey%5D=${shareAttributes.attributes[0].key}&attributes%5B0%5D%5Bvalue%5D=${shareAttributes.attributes[0].value}&attributes%5B1%5D%5Bscope%5D=${shareAttributes.attributes[1].scope}&attributes%5B1%5D%5Bkey%5D=${shareAttributes.attributes[1].key}&attributes%5B1%5D%5Bvalue%5D=${shareAttributes.attributes[1].value}`
+        body: 'shareType=0&shareWith=' + shareeName + '&path=%2F' + testFile + '&attributes%5B0%5D%5Bscope%5D=' + shareAttributes.attributes[0].scope +
```
Comment (individual-it) [inline tests/sharingWithAttributesTest.js:54]: can we make the lines even shorter?

**Review by individual-it — APPROVED** [review]
just small comments, you can decide if its worth fixing

**Review by individual-it — APPROVED** [review]

**Pull request merged**